include/mata/nfa.hh

```diff
@@ -684,6 +891,135 @@ std::ostream& operator<<(std::ostream& strm, const Nfa& nfa);
 /// global constructor to be called at program startup (from vm-dispatch)
 void init();
 
+/**
+ * Operations on segment automata.
+ */
+namespace SegNfa
```

Comment:
Is the new namespace necessary?

Reply:
It is not. However, after discussing the operations within this namespace with @vhavlena, we agreed that all of the operations are only applicable on a so-called segment automata (as described in the code). These operations will not (and cannot) work on all NFAs. Therefore, if a user wants to call these operations, they should be well aware they are working with different type of NFAs and of the significance of segment automata as parameters for these operations.

This way, the user cannot call any of the functions within `Mata::Nfa::SegNfa` without explicitly requesting operations on segment automata (by calling the operations as `SegNfa::operation()`). The same applies in #41 to function `noodlify`, for which apply the same constraints.

I believe this structure is sound. If we do not agree on that, it is no problem whatsoever to remove the inner namespace `SegNfa` and keep all functions within the general `Mata::Nfa` namespace. If we agree on keeping `SegNfa` namespace, do you think I should better explain (probably here in the namespace declaration) why is there a separate namespace and what operations does it contain?

Reply:
I think using new namespace is sound. Segment automata are a little bit different and it is ok to separate functions related to them in their own space. I would keep it as it is.

Reply:
Due to a lack of objections, I will keep the namespace as it is and mark this conversation as resolved. If we want to discuss this further, feel free to let me know.